qutip/visualization.py

```diff
@@ -170,6 +171,199 @@ def _cb_labels(left_dims):
+           note that this generalizes `"threshold"` to complex numbers.
+        -  If set to ``"phase"``, each color is chosen according to
+           the angle of the corresponding matrix element.
+
+    figure : a matplotlib Figure instance
+        The Figure canvas in which the plot will be drawn.
+
+    figure_options : dict
+        dict of options to create a figure object
+
+    axes : a matplotlib axes instance
+        The axes context in which the plot will be drawn.
+
+    axes_options : dict
+        dict of options to set to an axes object
+
+    Returns
+    -------
+    fig, ax : tuple
+        A tuple of the matplotlib figure and axes instances used to produce
+        the figure.
+
+    Raises
+    ------
+    ValueError
+        Input argument is not a quantum object.
+
+    Examples
+    --------
+    >>> import qutip
+    >>>
+    >>> dm = qutip.rand_dm(4)
+    >>> fig, ax = qutip.hinton(dm)
+    >>> fig.show()
+    >>>
+    >>> qutip.settings.colorblind_safe = True
+    >>> fig, ax = qutip.hinton(dm, color_style="threshold")
+    >>> fig.show()
+    >>> qutip.settings.colorblind_safe = False
+    >>>
+    >>> fig, ax = qutip.hinton(dm, color_style="phase")
+    >>> fig.show()
+    """
+
+    cmap = axes_options.pop('cmap', None)
+    if cmap is None:
+        cmap = color.cmap
+
+    if figure is None:
+        if axes is None:
+            figure, axes = plt.subplots(1, 1, **figure_options)
+        else:
+            figure = axes.get_figure()
+    else:
+        if axes is None:
+            axes = figure.add_subplots(1, 1)
+
+    axes.set(**axes_options)
+
+    # Extract plotting data W from the input.
+    if isinstance(rho, Qobj):
+        if rho.isoper:
+            W = rho.full()
+
+            # Create default labels if none are given.
+            if xlabels is None or ylabels is None:
+                labels = _cb_labels(rho.dims[0])
+                xlabels = xlabels if xlabels is not None else list(labels[0])
+                ylabels = ylabels if ylabels is not None else list(labels[1])
+
+        elif rho.isoperket:
+            W = vector_to_operator(rho).full()
+        elif rho.isoperbra:
+            W = vector_to_operator(rho.dag()).full()
+        elif rho.issuper:
+            if not isqubitdims(rho.dims):
+                raise ValueError("Hinton plots of superoperators are "
+                                 "currently only supported for qubits.")
+            # Convert to a superoperator in the Pauli basis,
+            # so that all the elements are real.
+            sqobj = _to_superpauli(rho)
+            nq = int(log2(sqobj.shape[0]) / 2)
+            W = sqobj.full().T
+            # Create default labels, too.
+            if (xlabels is None) or (ylabels is None):
+                labels = list(map("".join, it.product("IXYZ", repeat=nq)))
+                xlabels = xlabels if xlabels is not None else labels
+                ylabels = ylabels if ylabels is not None else labels
+
```

Comment:
When going over the old function, please try to clean then and fix issues, etc.

Here labels will not be generated it and operator ket (column stacked density matrix) is passed.
It also raise an error when `rho` is a super operator that is not made of qubits, which should be included in the docstring or generalized.

There are probably many issues like this. You can only leave comment (`#TODO : Add label`) if there are too many to fix.